posthog/session_recordings/session_recording_api.py

```diff
@@ -1001,32 +998,57 @@ def _stream_blob_v2_to_client(
             if max_blob_key >= len(blocks):
                 raise exceptions.NotFound("Block index out of range")
 
-            decompressed_blocks = []
-            with timer("fetch_block__stream_blob_v2_to_client"):
-                for block_index in range(min_blob_key, max_blob_key + 1):
+            async def fetch_single_block_async(block_index: int) -> tuple[int, str | None]:
+                try:
                     block = blocks[block_index]
-                    try:
-                        decompressed_blocks.append(session_recording_v2_object_storage.client().fetch_block(block.url))
-                    except BlockFetchError:
-                        logger.exception(
-                            "Failed to fetch block",
-                            recording_id=recording.session_id,
-                            team_id=self.team.id,
-                            block_index=block_index,
-                        )
-                        raise exceptions.APIException("Failed to load recording block")
+                    content = await asyncio.to_thread(
+                        session_recording_v2_object_storage.client().fetch_block, block.url
+                    )
+                    return block_index, content
+                except BlockFetchError:
+                    logger.exception(
+                        "Failed to fetch block",
+                        recording_id=recording.session_id,
+                        team_id=self.team.id,
+                        block_index=block_index,
+                    )
+                    return block_index, None
+
+            with timer("fetch_blocks_parallel__stream_blob_v2_to_client"):
```

Comment:
Nitpick: it's still async, so not exactly parallel? :)

Reply:
ooh interesting nitpick... because async is just not waiting, rather than necessarily running at the same time?